exercises/practice/paasio/src/test/java/PaasioTest.java

```diff
@@ -0,0 +1,337 @@
+import org.junit.jupiter.api.*;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+
+import static org.assertj.core.api.Assertions.*;
+
+public class PaasioTest {
+
+    private ByteArrayInputStream dataInputStream;
+    private ByteArrayOutputStream dataOutputStream;
+    private static final String MESSAGECONSTANT = "This is additional Content.";
+
+    @BeforeEach
+    public void setUPTest(TestInfo testInfo) {
+
+        this.dataInputStream = new ByteArrayInputStream("This is data ".getBytes());
+        this.dataOutputStream = new ByteArrayOutputStream();
+    }
+
+    @Test
+    void checkReadOperationCount() {
+
+        try (Paasio customFileReader = new Paasio(this.dataInputStream, this.dataOutputStream)) {
+            customFileReader.read();
+            customFileReader.read();
+            customFileReader.read();
+            assertThat(customFileReader.getReadOperationCount()).isEqualTo(3);
+        } catch (IOException ioException) {
+            System.out.println("Exception occured");
+        }
+
+    }
```

Comment:
Don't catch the exception in the tests, othewise the tests will pass if an exception is thrown. Instead, let propagate the exception so that JUnit also sees the exception.

```suggestion
    void checkReadOperationCount() throw Exception {

        try (Paasio customFileReader = new Paasio(this.dataInputStream, this.dataOutputStream)) {
            customFileReader.read();
            customFileReader.read();
            customFileReader.read();
            assertThat(customFileReader.getReadOperationCount()).isEqualTo(3);
        }
    }
```